UnityProject/Assets/Scripts/Items/Weapons/MagazineBehaviour.cs

```diff
@@ -38,12 +38,9 @@ public class MagazineBehaviour : NetworkBehaviour, IServerSpawn, IExaminable, IC
 	private double[] RNGContents;
 
 	/// <summary>
-	///	Whether this can be used to reload other (internal or external) magazines.
+	///	The type of magazine. This effects various behaviours depending on its setting
 	/// </summary>
-	[HideInInspector]
-	public bool isClip = false;
-	[NonSerialized]
-	public bool isCell = false;
+	public MagType magType;
```

Comment:
make it private with a public read only. Otherwise other scripts are able to change your magtype which I assume is not intended.

Reply:
its only public because eletrical magazine forces it to be a cell.
dont think i would be able to do that otherwise

Reply:
Uhmmm, do you get it outside of the class/children ? If not, it could be ``protected``

Reply:
gun.cs needs to get it, as mentioned previously